File: app/src/main/java/com/fishweeb/practical/TextButtonEntity.java
package com.fishweeb.practical;

import android.graphics.Color;

public class TextButtonEntity extends TextEntity
{
    protected String nextState;


    @Override
    public void OnClickFunction()
    {
        paint.setColor(onclickColor);
        StateManager.Instance.ChangeState(nextState);
    }

    @Override
    public void OffClickFunction()
    {
        paint.setColor(defaultColor);
    }

    public static TextButtonEntity Create(float xPos,float yPos,String text,float textSize,String _nextState)
    {
        TextButtonEntity result = new TextButtonEntity();
        result.Pos.x = xPos;
        result.Pos.y = yPos;
        result.SetText(text);
        result.nextState = _nextState;
        result.paint.setTextSize(textSize);
        EntityManager.Instance.AddEntity(result);
        return result;
    }
}

File: app/src/main/java/edu/androidgroupproject/TextButtonEntity.kt
package edu.androidgroupproject

class TextButtonEntity : TextEntity() {
    protected var nextState: String? = null


    override fun OnClickFunction() {
        paint.color = onclickColor
        StateManager.Companion.Instance.ChangeState(nextState)
    }

    override fun OffClickFunction() {
        paint.color = defaultColor
    }

    companion object {
        fun Create(
            xPos: Float,
            yPos: Float,
            text: String?,
            textSize: Float,
            _nextState: String?
        ): TextButtonEntity {
            val result = TextButtonEntity()
            result.Pos.x = xPos
            result.Pos.y = yPos
            result.SetText(text)
            result.nextState = _nextState
            result.paint.textSize = textSize
            EntityManager.Companion.Instance.AddEntity(result)
            return result
        }
    }
}
